test/parallel/test-net-localerror.js

```diff
@@ -20,25 +20,24 @@
 // USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 'use strict';
-require('../common');
-const assert = require('assert');
+const common = require('../common');
 const net = require('net');
 
-// Using port 0 as localPort / localAddress is already invalid.
+const connect = (opts, code, type) => {
+  common.expectsError(
+    () => net.connect(opts),
+    { code, type }
+  );
+};
+
 connect({
   host: 'localhost',
   port: 0,
-  localPort: 'foobar',
-}, /^TypeError: "localPort" option should be a number: foobar$/);
+  localAddress: 'foobar',
+}, 'ERR_INVALID_IP_ADDRESS', Error);
```

Comment:
This was a `TypeError` before?

Reply:
my bad. I corrected this, now the error types in lib/net.js have the correct value.